File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    
    double root = sqrt(n);
    if(pow(ceil(root), 2) == n)
        answer = pow(root + 1, 2);
    else
        answer = -1;
    
    return answer;
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
using System;

public class Solution {
    public long solution(long n) {
        long answer = 0;
        
        double num = Math.Sqrt(n);
        Console.WriteLine(num);
        if(Math.Pow(Math.Ceiling(num), 2) != n)
            answer = -1;
        else
            answer = (long)Math.Pow(num + 1, 2);
        
        return answer;
    }
}